string readFile
string writeFile
---
string weight
